modular_skyrat/modules/hev_suit/code/pcv_suit.dm

```diff
@@ -0,0 +1,701 @@
+
+#define PCV_ARMOR_POWEROFF list(30, 30, 30, 30, 30, 30, 20, 20, 20, 10)
+
+#define PCV_ARMOR_POWERON list(40, 40, 40, 40, 40, 40, 50, 50, 50, 30)
+
+#define PCV_POWERUSE_AIRTANK 2
+#define PCV_POWERUSE_HIT 100
+#define PCV_POWERUSE_HEAL 150
+
+#define PCV_COOLDOWN_HEAL 10 SECONDS
+#define PCV_COOLDOWN_RADS 20 SECONDS
+#define PCV_COOLDOWN_ACID 20 SECONDS
+
+#define PCV_HEAL_AMOUNT 5
+#define PCV_BLOOD_REPLENISHMENT 20
+
+#define PCV_NOTIFICATION_TEXT_AND_VOICE "VOICE_AND_TEXT"
+#define PCV_NOTIFICATION_TEXT "TEXT_ONLY"
+#define PCV_NOTIFICATION_VOICE "VOICE_ONLY"
+#define PCV_NOTIFICATION_OFF "OFF"
+#define PCV_NOTIFICATIONS list(PCV_NOTIFICATION_TEXT_AND_VOICE, PCV_NOTIFICATION_TEXT, PCV_NOTIFICATION_VOICE, PCV_NOTIFICATION_OFF)
+
+/obj/item/clothing/head/helmet/space/pcv_suit
+	name = "powered combat helmet"
+	desc = "The Mark II PCV suit helmet."
+	icon = 'modular_skyrat/modules/awaymissions_skyrat/icons/hecucloth.dmi'
+	worn_icon = 'modular_skyrat/modules/awaymissions_skyrat/icons/hecumob.dmi'
+	icon_state = "hecu_helm"
+	inhand_icon_state = "sec_helm"
+	armor = list(MELEE = 30, BULLET = 30, LASER = 30, ENERGY = 30, BOMB = 30, BIO = 20, FIRE = 20, ACID = 20, WOUND = 10)
+	flags_inv = HIDEHAIR
+	obj_flags = NO_MAT_REDEMPTION
+	resistance_flags = FIRE_PROOF|ACID_PROOF|FREEZE_PROOF
+	clothing_flags = SNUG_FIT
+	clothing_traits = list(TRAIT_MEDICAL_HUD,TRAIT_NIGHT_VISION)
+	visor_flags_inv = HIDEMASK|HIDEEYES|HIDEFACE|HIDEFACIALHAIR|HIDESNOUT
+	visor_flags = null
+	slowdown = 0
+
+/obj/item/clothing/suit/space/pcv_suit
+	name = "powered combat vest"
+	desc = "The Mark II PCV suit protects the user from a number of hazardous environments and has in build ballistic protection."
+	icon = 'modular_skyrat/modules/awaymissions_skyrat/icons/hecucloth.dmi'
+	worn_icon = 'modular_skyrat/modules/awaymissions_skyrat/icons/hecumob.dmi'
+	icon_state = "hecu_vest"
+	armor = list(MELEE = 30, BULLET = 30, LASER = 30, ENERGY = 30, BOMB = 30, BIO = 20, FIRE = 20, ACID = 20, WOUND = 10) //This is muhreens suit, of course it's mildly strong and balanced for PvP.
+	flags_inv = null
+	allowed = list(/obj/item/gun, /obj/item/ammo_box,/obj/item/ammo_casing, /obj/item/melee/baton, /obj/item/melee/energy/sword, /obj/item/restraints/handcuffs, /obj/item/tank/internals)
+	cell = /obj/item/stock_parts/cell/hyper
+	actions_types = list(/datum/action/item_action/pcv_toggle, /datum/action/item_action/pcv_toggle_notifs)
+	resistance_flags = FIRE_PROOF|ACID_PROOF|FREEZE_PROOF
+	clothing_flags = SNUG_FIT
+
+	var/activated = FALSE
+	var/activating = FALSE
+
+	var/mob/living/carbon/current_user
+	var/obj/item/clothing/head/helmet/space/pcv_suit/current_helmet
+	var/obj/item/tank/internals/current_internals_tank
+	var/obj/item/radio/internal_radio
+
+	var/user_old_bruteloss
+	var/user_old_fireloss
+	var/user_old_toxloss
+	var/user_old_cloneloss
+	var/user_old_oxyloss
+
+	var/radio_channel = RADIO_CHANNEL_COMMON
+
+	var/timer_id = null
+
+	var/voice_current_cooldown
```

Comment:
don't use world.time cooldowns, use the TG `DECLARE_COOLDOWN` code instead

Reply:
No clue what this is.